enhancements/network/sdn-live-migration.md

```diff
@@ -0,0 +1,331 @@
+---
+title: SDN Live Migration
+authors:
+  - "@pliurh"
+reviewers:
+  - "@danwinship"
+  - "@trozet"
+  - "@dcbw"
+approvers:
+  - 
+creation-date: 2022-03-04
+last-updated: 2022-03-04
+status: implementable
+---
+
+# SDN Live Migration
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+
+## Summary
+Migrating the CNI network provider network of a running cluster from
+OpenShift SDN to OVN Kubernetes without service interruption. During the
+migration, we will partition the cluster into two sets of nodes controlled by
+different network plugins. We will utilize the Hybrid overlay feature of
+OVN Kubernetes to connect the networks of the two CNI network plugins. So that
+pods on each side can still talk to the pods on the other side.
+
+## Motivation
+
+For some Openshift users, they have very high requirement on service
+availability. The current SDN migration solution, that will cause a service
+interruption, is not acceptable.
+
+### Goals
+
+- Migrate the cluster network provider from OpenShift SDN to OVN Kubernetes for a 
+  existing cluster.
+- This's in-place migration without requiring extra nodes.
+- The impact to workload of the migration shall be similar as an OCP upgrade.
+- The solution can work in scale, e.g. in a large cluster with hundreds of
+  nodes.
+- The migration operation shall be able to be rolled back if needed.
+
+### Non-Goals
+
+- Support for migration to other network providers
+- The necessary GUI change in Openshift Cluster Manager.
+- The live migration of the egress IP, router and firewall configuration.
+- Migration from SDN Multitenant mode
+
+## Pre-requisites
+
+- This solution relies on the ovnkube hybrid overlay feature. This bug 
+  https://bugzilla.redhat.com/show_bug.cgi?id=2040779 needs to be fixed.
+
+## Proposal
+
+The key problem of do a live SDN migration is that, we need to the connectivity 
+of the cluster network during the migration, when pods are attached to different
+networks. We propose to utilize the OVN Kubernetes hybrid overlay feature to
+connect the networks owned by OpenShift SDN and OVN Kubernetes.
+
+- We will run different plugins on different nodes, but both plugins will know
+  how to reach pods owned by the other plugin, so all pods/services/etc remain
+  connected.
+- During migration, CNO will take original-plugin nodes one by one and convert
+  them to destination-plugin nodes, rebooting them in the process.
+- The cluster network CIDR will not change. In fact, none of the node host
+  subnets will change.
+- NetworkPolicy will work correctly throughout the migration.
+
+### Limitations
```

Comment:
Maybe describe the features called out in "non-goals" in this section in more detail?

Reply:
added more description below. PTAL

Reply:
Thanks.

What if the implementation in the CNO did validations before any migration begins?  It could check the cluster for anything it knows may not work?  I know some things are difficult to check for (like multicast traffic), but some things should be relatively straight forward if it's just looking at the kube API.

Reply:
I think it is a good idea. I added it to the `Migration Setup` section. Thanks.